src/Microsoft.Azure.SignalR.Management/Configuration/ServiceManagerOptions.cs

```diff
@@ -62,26 +57,18 @@ private void ValidateServiceEndpoint()
             {
                 notNullCount += 1;
             }
-            if (ServiceEndpoint != null)
-            {
-                notNullCount += 1;
-            }
-            if (ServiceEndpoints != null)
+            if (Endpoints != null)
             {
                 notNullCount += 1;
             }
 
             if (notNullCount == 0)
             {
-                throw new InvalidOperationException($"Service endpoint(s) is/are not configured. Please set one of the following properties {nameof(ConnectionString)}, {nameof(ServiceEndpoint)}, {nameof(ServiceEndpoints)}.");
-            }
-            if (notNullCount > 1)
-            {
-                throw new InvalidOperationException($"Please set ONLY one of the following properties: {nameof(ConnectionString)}, {nameof(ServiceEndpoint)}, {nameof(ServiceEndpoints)}.");
+                throw new InvalidOperationException($"Service endpoint(s) is/are not configured. Please set one of the following properties {nameof(ConnectionString)}, {nameof(Endpoints)}.");
             }
-            if (ServiceEndpoints != null && ServiceEndpoints.Length == 0)
+            if (ConnectionString == null && Endpoints != null && Endpoints.Length == 0)
```

Comment:
looks like you allow both `ConnectionString` set and `Endpoints` set now? Then there is no need to do check here, instead, you can check if the merged endpoints is empty?